integration/non_root_user_cap_test.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright The containerd Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+func TestNonRootUserCap(t *testing.T) {
+	for name, test := range map[string]struct {
+		uid     int64
+		startOK bool
+	}{
+		"shouldn't be able to run container with a private workdir only accessible to nobody as a non-root user": {
```

Comment:
suggest: "shouldn't be able to run test container, with a private workdir that is accessible to nobody, as non-root user"